### PR TITLE
Update licensing docs to account for virtual agents.

### PR DESCRIPTION
**Pull request by mkeeler**
Updating our licensing docs to describe how consul-dataplane and virtual agents impact pricing/licensing.

### CODE REVIEW

**Review by trujillo-adam — COMMENTED** [review]
Made some suggestions to try to clarify the language.

**Review by ishustava — APPROVED** [review]
LGTM! Thanks for doing it Matt!

File: website/content/docs/enterprise/license/overview.mdx [hunk 1]
```diff
@@ -20,6 +20,10 @@ agent's configuration or environment. Also, prior to 1.10.0, server agents would
 the license between themselves. This no longer occurs and the license must be present on each server agent
 when it is started.
 
+Consul Enterprise 1.14.0, when running on Kubernetes, removed client agents and replaced these with virtual agents.
+Virtual agents represent nodes on which Consul service mesh services are running and are used for the purposes of 
+counting license entitlements and related billing in conjunction with node-based licensing and/or pricing.
```
[the remark this answers is not in the document]
Comment (trujillo-adam) [reply website/content/docs/enterprise/license/overview.mdx:25]: @mkeeler - this is what we agreed on when I spoke to legal

**Review by trujillo-adam — COMMENTED** [review]
met real quick with legal and came up with appropriate language

**Pull request merged**